lib/rexml/parsers/baseparser.rb

```diff
@@ -471,9 +471,13 @@ def pull_event
                 end
 
                 return [ :comment, md[1] ]
-              else
-                md = @source.match(/\[CDATA\[(.*?)\]\]>/um, true)
-                return [ :cdata, md[1] ] if md
+              elsif @source.match?("[CDATA[", true)
+                text = @source.read_until("]]>")
+                if text.chomp!("]]>")
+                  return [ :cdata, text ]
+                else
+                  raise REXML::ParseException.new("Malformed CDATA", @source)
```

Comment:
Can we add what is wrong for easy to debug? Something like:

```suggestion
                  raise REXML::ParseException.new("Malformed CDATA: Missing end ']]>'", @source)
```

Reply:
Thanks.
Done.